src/detectors/DockerCGroupV1Detector.ts

```diff
@@ -0,0 +1,130 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* This is based on a detector from OTel https://github.com/open-telemetry/opentelemetry-js/blob/main/packages/opentelemetry-resources/src/detectors/
+ We're copying this code and changing the implementation to a synchronous one from async. This is required for our distribution to not incur ~1 second of overhead
+ when setting up the tracing pipeline. This is a temporary solution until we can agree upon and implement a solution upstream.
+*/
+
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Resource, ResourceDetectionConfig } from '@opentelemetry/resources';
+
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions';
+
+import * as fs from 'fs';
+import { diag } from '@opentelemetry/api';
+
+const isValidBase16String = (hexString: string) => {
+  for (let ch = 0; ch < hexString.length; ch++) {
+    const code = hexString.charCodeAt(ch);
+    if (
+      (48 <= code && code <= 57) ||
+      (97 <= code && code <= 102) ||
+      (65 <= code && code <= 70)
+    ) {
+      continue;
+    }
+    return false;
+  }
+  return true;
+};
+
+export class DockerCGroupV1Detector {
+  public detect(_config?: ResourceDetectionConfig): Resource {
+    try {
+      const containerId = this._getContainerId();
+      return !containerId
+        ? Resource.empty()
+        : new Resource({
+            [SemanticResourceAttributes.CONTAINER_ID]: containerId,
+          });
+    } catch (e) {
+      diag.info(
+        'Docker CGROUP V1 Detector did not identify running inside a supported docker container, no docker attributes will be added to resource: ',
+        e
+      );
+      return Resource.empty();
+    }
+  }
+
+  protected _getContainerId(): string | null {
+    try {
+      const rawData = fs.readFileSync('/proc/self/cgroup', 'utf8').trim();
+      return this._parseFile(rawData);
+    } catch (e) {
+      if (e instanceof Error) {
+        const errorMessage = e.message;
+        diag.info(
```

Comment:
Yes, with the current implementation it would. Your suggestion?

How about we detect the platform and only call `_getContainerId()` if it's linux?

Reply:
I guess the whole detector could check for `process.platform === 'linux'`

Reply:
Not sure what you mean by "whole". Since it's currently the only platform-specific detector, added it into the detector itself. Once we have more, we could do it in `resource.ts` to optimize.